src/gui/sdlmain.cpp

```diff
@@ -2209,6 +2229,109 @@ static SDL_Point window_bounds_from_label(const std::string &pref,
 	return {w, h};
 }
 
+static SDL_Point clamp_to_minimum_window_dimensions(SDL_Point size)
+{
+	const auto w = std::max(size.x, FALLBACK_WINDOW_DIMENSIONS.x);
+	const auto h = std::max(size.y, FALLBACK_WINDOW_DIMENSIONS.y);
+	return {w, h};
+}
+
+static SDL_Rect get_desktop_resolution()
+{
+	SDL_Rect desktop;
+	assert(sdl.display_number >= 0);
+	SDL_GetDisplayBounds(sdl.display_number, &desktop);
+	assert(desktop.w >= FALLBACK_WINDOW_DIMENSIONS.x);
+	assert(desktop.h >= FALLBACK_WINDOW_DIMENSIONS.y);
+	return desktop;
+}
+
+// Takes in:
+//  - The 'maxresolution' config value: 'auto', 'W,H', or an invalid setting.
+//
+// Except for SURFACE and TEXTURE rendering, the function populates the following struct members:
+//  - 'sdl.desktop.restrict_resolution', true if resolution restriction is enabled.
+//  - 'sdl.desktop.max_resolution', with the refined size.
+
+static void setup_resolution_restriction_from_conf(const std::string &max_resolution_val)
+{
+	sdl.restrict_resolution = false;
```

Comment:
You're right, it's best to avoid ambiguity in the long run.

I ended up with `max_resolution` because it's consistent with the existing namings (`window_resolution`, `viewport_resolution`, etc.), and `use_max_resolution` for the flag. Changed all the other ambigous namings in the code & comments.